chain/chain/src/store.rs

```diff
@@ -1750,6 +1739,19 @@ impl<'a> ChainStoreUpdate<'a> {
         }
     }
 
+    /// Save header head in Epoch Sync
+    /// Checking validity of header head is delegated to Epoch Sync methods
+    pub fn save_header_head_forcibly(&mut self, t: &Tip) -> Result<(), Error> {
+        self.try_save_latest_known(t.height)?;
+
+        self.chain_store_cache_update.height_to_hashes.insert(t.height, Some(t.last_block_hash));
```

Comment:
It seems that `height_to_hashes` is used to update `ColBlockHeight`, which stores blocks, not block headers, by height. Therefore I wonder whether this line here breaks some invariant since now we potentially don't have the corresponding block in storage.